src/stringobj.h

```diff
@@ -296,16 +356,22 @@ Obj ConvImmString(Obj str);
 **
 ** The cstring is assumed to be allocated on the heap, hence its length
 ** is dynamic and must be computed during runtime using strlen.
+**
+** This macro is provided for backwards compatibility of packages.
+** MakeString and MakeImmString are as efficient and should be used instead.
 */
 #define C_NEW_STRING_DYN(string,cstr) \
   C_NEW_STRING(string, strlen(cstr), cstr)
 
 /****************************************************************************
 **
-*F  C_NEW_STRING_CONST( <string>, <cstring> ) . . . . . . . . create GAP string
+*F  C_NEW_STRING_CONST( <string>, <cstring> ) . . . . . . create GAP string
```

Comment:
As far as I can tell, no package uses `C_NEW_STRING_CONST`, so you could just as well remove this.